### DISCUSSION TITLE
Server-side service, browser flows and CSRF cookies problem

### DISCUSSION THREAD

**Discussion by InverMN** [ory/kratos#1556] [Q&A] [answered]
Hi, last time I upgraded Kratos from v0.6 to v0.7 and my server-side self-service stopped working. After a search, I found on the Kratos Github Releases page, that `<kratos-admin>/self-service/*/flows` endpoints have been deleted.

`If a self-service flow for browsers is requested, the CSRF cookie must be included in the call, regardless if it is a client-side browser app or a server-side browser app calling.`

But in [docs](https://www.ory.sh/kratos/docs/debug/csrf/#accessing-ory-kratos-apis-from-a-server-side-application), there is written there's no need for providing CSRF cookie and self-service should use admin endpoints.

How should I interpret these?

**Comment by aeneasr** [comment, marked as answer]
That section is outdated, we need to update it. Please see the changelog for 0.7: https://github.com/ory/kratos/blob/master/CHANGELOG.md#070-alpha1-2021-07-13

**Reply by InverMN** [reply]
So... I must get CSRF cookie? That will be kkinda painful workflow, I think

**Reply by aeneasr** [reply]
If your UI and Ory Kratos are on the same domain, it's really not a problem. Check out how we solved it:

https://github.com/ory/kratos-selfservice-ui-node/blob/782f20cb023c140b230dfa0f5358c76999e9b938/src/routes/login.ts#L25

We also updated documentation on more examples etc:

https://www.ory.sh/kratos/docs/self-service/flows/user-login

More context: https://github.com/ory/kratos/issues/1282
